src/nupic/algorithms/AnomalyLikelihood.cpp

```diff
@@ -0,0 +1,340 @@
+      filteredLikelihoods[i] = yellowThreshold;
+    }
+  }
+
+  NTA_ASSERT(filteredLikelihoods.size()==likelihoods.size());
+  return filteredLikelihoods;
+}
+
+
+vector<Real>  AnomalyLikelihood::updateAnomalyLikelihoods_(vector<Real> anomalyScores, UInt verbosity) { 
+  if (verbosity > 3) {
+    cout << "In updateAnomalyLikelihoods."<< endl;
+    cout << "Number of anomaly scores: "<<  anomalyScores.size() << endl;
+//    cout << "First 20:", anomalyScores[0:min(20, len(anomalyScores))])
+    cout << "Params: name=" <<  distribution_.name << " mean="<<distribution_.mean <<" var="<<distribution_.variance <<" stdev="<<distribution_.stdev <<endl;
+  }
+
+ NTA_CHECK(anomalyScores.size() > 0); // "Must have at least one anomalyScore"
+
+  // Compute moving averages of these new scores using the previous values
+  // as well as likelihood for these scores using the old estimator 
+  vector<Real> likelihoods;
+  likelihoods.reserve(runningAverageAnomalies_.size()); 
+  for (auto newAverage : runningAverageAnomalies_) { //TODO we could use transform() here (? or would it be less clear?) 
+    likelihoods.push_back(tailProbability_(newAverage)); 
+  }
+
+  // Filter the likelihood values. First we prepend the historical likelihoods
+  // to the current set. Then we filter the values.  We peel off the likelihoods
+  // to return and the last windowSize values to store for later.
+  UInt toCrop = min((UInt)this->averagedAnomaly_.getMaxWindowSize(), (UInt)runningLikelihoods_.size());
+  this->runningLikelihoods_.insert(runningLikelihoods_.end() - toCrop, likelihoods.begin(),likelihoods.end()); //append & crop
+  cerr << "FUU " << this->runningLikelihoods_.size() << "  <= " << this->averagedAnomaly_.getMaxWindowSize() << endl; 
+//!  NTA_CHECK(this->runningLikelihoods_.size() <= this->averagedAnomaly_.getMaxWindowSize()); //FIXME fix this check, returns ~"14xx <= 10"
+
+  auto filteredLikelihoods = filterLikelihoods_(likelihoods);
+
+  if (verbosity > 3) {
+    cout << "Number of likelihoods:"<< likelihoods.size() << endl;
+//    print("First 20 likelihoods:", likelihoods[0:min(20, len(likelihoods))])
+    cout << "Leaving updateAnomalyLikelihoods."<< endl;
+  }
+
+  return filteredLikelihoods;
+}
+
+
+vector<Real> AnomalyLikelihood::estimateAnomalyLikelihoods_(vector<Real> anomalyScores, UInt skipRecords, UInt verbosity) { //FIXME averagingWindow not used, I guess it's not a sliding window, but aggregating window (discrete steps)!
+  if (verbosity > 1) {
+    cout << "In estimateAnomalyLikelihoods_."<<endl;
+    cout << "Number of anomaly scores:" <<  anomalyScores.size() << endl;
+    cout << "Skip records="<<  skipRecords << endl;
+//    print("First 20:", anomalyScores[0:min(20, len(anomalyScores))])
+  }
+
+  NTA_CHECK(anomalyScores.size() > 0); // "Must have at least one anomalyScore"
+  auto dataValues = anomalyScores; //FIXME the "data" should be anomaly scores, or raw values? 
```

Comment:
This comment can be removed, I assume we deal with anomaly scores only here, right?